packages/compass-crud/src/components/bulk-delete-modal.tsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import {
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  Button,
+  TextInput,
+  KeylineCard,
+  css,
+  cx,
+  spacing,
+  InfoSprinkle,
+  Label,
+} from '@mongodb-js/compass-components';
+import ReadonlyDocument from './readonly-document';
+
+const modalFooterSpacingStyles = css({
+  gap: spacing[2],
+});
+
+const documentHorizontalWrapper = css({
+  display: 'flex',
+  flexDirection: 'row',
+  flex: 'none',
+  flexShrink: 0,
+  overflow: 'auto',
+  marginBottom: spacing[2],
+  gap: spacing[2],
+  maxWidth: '100%',
+});
+
+const documentContainerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  flex: 'none',
+  flexShrink: 0,
+  marginBottom: spacing[2],
+  width: '100%',
+});
+
+const documentStyles = css({
+  flexBasis: '164px',
+  flexGrow: 1,
+  flexShrink: 0,
+  overflow: 'auto',
+  padding: 0,
+  width: '100%',
+});
+
+const modalBodySpacingStyles = css({
+  marginTop: spacing[3],
+  paddingLeft: spacing[5],
+  display: 'flex',
+  flexDirection: 'column',
+  gap: spacing[3],
+});
+
+const previewStyles = css({
+  minHeight: '200px',
+});
+
+type QueryLabelProps = {
+  tooltip: string;
+  label: string;
+};
+
+const queryLabelStyles = css({
+  display: 'flex',
+  gap: spacing[2],
+  alignItems: 'center',
+});
+
+const QueryLabel: React.FunctionComponent<QueryLabelProps> = ({
+  tooltip,
+  label,
+}) => {
+  return (
+    <div className={queryLabelStyles}>
+      <Label htmlFor="template-dropdown">{label}</Label>
+      <InfoSprinkle align="right">{tooltip}</InfoSprinkle>
+    </div>
+  );
+};
+
+type BulkDeleteModalProps = {
+  open: boolean;
+  documentCount: number;
+  filterQuery: string;
+  namespace: string;
+  sampleDocuments: Document[];
+  onCancel: () => void;
+  onConfirmDeletion: () => void;
+};
+
+const BulkDeleteModal: React.FunctionComponent<BulkDeleteModalProps> = ({
+  open,
+  documentCount,
+  filterQuery,
+  namespace,
+  sampleDocuments,
+  onCancel,
+  onConfirmDeletion,
+}) => {
+  const preview = (
+    <div className={documentHorizontalWrapper}>
```

Comment:
The designs are not clear on this (horizontal scroll), but i struggled a little to find that its a horizontal scroll.

Reply:
When we had the discussion, the idea was to keep what the aggregation screen does, so we are keeping the horizontal scroll.